### PR TITLE
[14.0][MIG] point_of_sale: migration script

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Finishing https://github.com/OCA/OpenUpgrade/pull/3106.

### CODE REVIEW

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/noupdate_changes.xml [hunk 1]
```diff
@@ -1,27 +1,27 @@
 <?xml version='1.0' encoding='utf-8'?>
 <odoo>
   <record id="product_product_consumable" model="product.product">
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/noupdate_changes.xml]: Please comment whole record tag for avoiding unneeded writings.

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,103 @@
+---Models in module 'point_of_sale'---
+---Fields in module 'point_of_sale'---
+point_of_sale / pos.config               / iface_precompute_cash (boolean): DEL
+point_of_sale / pos.config               / manage_orders (boolean)       : NEW
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt]: I think we have to enable this by default on existing configs, for providing this new feature, specially if coming from `pos_order_mgmt`. We are working on the transition from this module for this version (it seems the return is not available, but the config says that it exists). cc @chienandalu

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+def fill_sesion_on_picking_by_pos_order(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE stock_picking
+        SET pos_session_id = (SELECT session_id
+                                FROM pos_order
+                                WHERE pos_order.id = stock_picking.pos_order_id
+                                LIMIT 1)
+        WHERE pos_session_id IS NULL AND pos_order_id IS NOT NULL;
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py]: Simplify this expression:

```suggestion
        UPDATE stock_picking sp
        SET pos_session_id =  po.session_id
        FROM pos_order po
        WHERE po.id = sp.pos_order_id
            AND sp.pos_session_id IS NULL
```

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+def transfer_new_session_state(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE pos_session ps
+        SET state = CASE
+                        WHEN pc.cash_control=True AND
+                        COALESCE(ps.rescue,False)=False THEN 'opening_control'
+                        ELSE 'opened'
+                    END
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py]: Make it more readable:

```suggestion
        UPDATE pos_session ps
        SET state = CASE
           WHEN pc.cash_control AND NOT COALESCE(ps.rescue, False)
           THEN 'opening_control'
           ELSE 'opened'
       END
```

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+from openupgradelib import openupgrade
+
+
+def convert_field_m2o_to_o2m(env):
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py]: Better method name?

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/pre-migration.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+def fill_pos_config_default_picking_type_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE pos_config
+        SET picking_type_id = (SELECT pos_type_id
+                                FROM stock_warehouse
+                                WHERE stock_warehouse.company_id = pos_config.company_id
+                                LIMIT 1)
+        WHERE picking_type_id IS NULL;
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/pre-migration.py]:
```suggestion
        UPDATE pos_config pc
        SET picking_type_id = sw.pos_type_id
        FROM stock_warehouse sw
        WHERE sw.company_id = pc.company_id
            AND pc.picking_type_id IS NULL
```

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt (hunk 2 above)
Comment (chienandalu) [reply openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/upgrade_analysis_work.txt]: Yes, it says it exists but it doesn't. For point_of_sale, v14 is an advance of the functionallity that landed in v15. Indeed, this field doesn't exist in that version anymore, as they backed away from this approach (and in v16 the UI flow is similar although the whole module got a refactor.

In a summary I'd be setting it on by default as it's going to be in future versions.

File: openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py (hunk 3 above)
Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/point_of_sale/14.0.1.0.1/post-migration.py]: In fact, when I made the PR I didn't checked if the queries could be simplified. Thanks

**Pull request merged**